packages/components/src/components/telekom/telekom-header-data-back-compat/telekom-header-data-back-compat.tsx

```diff
@@ -0,0 +1,384 @@
+                                            if (
+                                              typeof grandChild.onClick ===
+                                              'function'
+                                            ) {
+                                              grandChild.onClick(event);
+                                            }
+                                          }}
+                                        >
+                                          {grandChild.name}
+                                        </a>
+                                      </li>
+                                    );
+                                  })}
+                                </ul>
+                              )}
+                            </scale-telekom-mega-menu-column>
+                          );
+                        })}
+                      </scale-telekom-mega-menu>
+                    </scale-telekom-nav-flyout>
+                  )}
+                </scale-telekom-nav-item>
+              );
+            })}
+          </scale-telekom-nav-list>
+        )}
+
+        {!readData(this.iconNavigation) ? null : (
+          <scale-telekom-nav-list
+            variant="functions"
+            slot="functions"
+            alignment="right"
+          >
+            {readData(this.iconNavigation)
+              .filter(({ id }) => id !== 'menu')
+              .map((item) => {
+                return (
+                  <scale-telekom-nav-item>
+                    <a
+                      href={item.href || 'javascript:void(0);'}
+                      onClick={(event) => {
+                        if (typeof item.onClick === 'function') {
+                          item.onClick(event);
+                        }
+                      }}
+                    >
+                      {item.badgeLabel ? (
+                        <scale-badge
+                          count={item.badgeCount}
+                          label={item.badgeLabel}
+                          label-visually-hidden
+                        >
+                          {renderIcon({
+                            tag: `scale-icon-${item.icon}`,
+                            attributes: {},
+                          })}
+                        </scale-badge>
+                      ) : (
+                        renderIcon({
+                          tag: `scale-icon-${item.icon}`,
+                          attributes: {},
+                        })
+                      )}
+                    </a>
+                  </scale-telekom-nav-item>
+                );
+              })}
+
+            <scale-telekom-nav-item class="burger-item">
+              <button>
+                <scale-badge>
+                  <scale-icon-action-menu></scale-icon-action-menu>
+                </scale-badge>
+              </button>
+              <scale-telekom-nav-flyout class="mobile-nav-flyout">
```

Comment:
These `burger-item` and `mobile-nav-flyout` classes are 💩 because people will need the code but it's "hidden", no? Please don't do anything about them now. 

I'll try and add an update to `scale-telekom-header` now (probably some news props) so we can remove them (here and in the .vue file where they're defined).

Reply:
The classes are my fault in the first place… 🫣